siriuspy/siriuspy/devices/orbit_interlock.py

```diff
@@ -101,25 +229,1349 @@ class BPMInterlock(_Device):
+        """
+        return self['IntlkAngBigger-Mon']
+
+    @property
+    def ang_inst_smaller(self):
+        """
+        Instantaneous angulation interlock set when either X or Y
+        minimum thresholds are exceeded.
+        """
+        return self['IntlkAngSmallerAny-Mon']
+
+    @property
+    def ang_inst_bigger(self):
+        """
+        Instantaneous angulation interlock set when either X or Y
+        maximum thresholds are exceeded.
+        """
+        return self['IntlkAngBiggerAny-Mon']
+
+    @property
+    def ang_inst_smaller_x(self):
+        """
+        Instantaneous angulation interlock set when X
+        minimum threshold is exceeded.
+        """
+        return self['IntlkAngSmallerX-Mon']
+
+    @property
+    def ang_inst_bigger_x(self):
+        """
+        Instantaneous angulation interlock set when X
+        maximum threshold is exceeded.
+        """
+        return self['IntlkAngBiggerX-Mon']
+
+    @property
+    def ang_inst_smaller_y(self):
+        """
+        Instantaneous angulation interlock set when Y
+        minimum threshold is exceeded.
+        """
+        return self['IntlkAngSmallerY-Mon']
+
+    @property
+    def ang_inst_bigger_y(self):
+        """
+        Instantaneous angulation interlock set when Y
+        maximum threshold is exceeded.
+        """
+        return self['IntlkAngBiggerY-Mon']
+
+    @property
+    def ang_latch_smaller(self):
+        """
+        Latch angulation interlock set when either X or Y
+        minimum thresholds are exceeded.
+        """
+        return self['IntlkAngSmallerLtc-Mon']
+
+    @property
+    def ang_latch_bigger(self):
+        """
+        Latch angulation interlock set when either X or Y
+        maximum thresholds are exceeded.
+        """
+        return self['IntlkAngBiggerLtc-Mon']
+
+    @property
+    def ang_latch_smaller_x(self):
+        """
+        Latch angulation interlock set when X
+        minimum threshold is exceeded.
+        """
+        return self['IntlkAngSmallerLtcX-Mon']
+
+    @property
+    def ang_latch_bigger_x(self):
+        """
+        Latch angulation interlock set when X
+        maximum threshold is exceeded.
+        """
+        return self['IntlkAngBiggerLtcX-Mon']
+
+    @property
+    def ang_latch_smaller_y(self):
+        """
+        Latch angulation interlock set when Y
+        minimum threshold is exceeded.
+        """
+        return self['IntlkAngSmallerLtcY-Mon']
+
+    @property
+    def ang_latch_bigger_y(self):
+        """
+        Latch angulation interlock set when Y
+        maximum threshold is exceeded.
+        """
+        return self['IntlkAngBiggerLtcY-Mon']
+
+
+class OrbitInterlock(BaseOrbitIntlk, _Devices):
+    """Orbit Interlock device."""
+
+    TIMEOUT = 10
+
+    # NOTE: this is a temporary location, will migrate to csconstants.
+    # The values will be replaced by the measured calibration data.
+    CONV_POLY_MONIT1_2_MONIT = _np.array([
```

Comment:
change the shape of the array to be the transposed of this one, so you can define the whole polynom using less lines of code.

Reply:
done!